From ac05d3469d5e3673f2a9746903b6b906810d6b73 Mon Sep 17 00:00:00 2001
From: Peter W Morreale <pmorreal@brocade.com>
Date: Fri, 21 Feb 2014 10:53:18 -0700
Subject: [PATCH 2/5] add mutually exclusive policy.

Make 'local ip' and 'local ip range' options are mutually exclusive.
They cannot co-exist, so make sure the user only configures one of them.

Signed-off-by: Peter W Morreale <pmorreal@brocade.com>
---
 file.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/file.c b/file.c
index 9f28d74..d3b49a1 100644
--- a/file.c
+++ b/file.c
@@ -980,6 +980,12 @@ int set_localiprange (char *word, char *value, int context, void *item)
                   word);
         return -1;
     }
+
+    if (lns->localaddr) {
+        snprintf (filerr, sizeof (filerr), "'local ip range' and 'local ip' are mutually exclusive\n");
+	return -1;
+    }
+
     lns->localrange = set_range (word, value, lns->localrange);
     if (!lns->localrange)
         return -1;
@@ -1073,6 +1079,11 @@ int set_localaddr (char *word, char *value, int context, void *item)
         return set_ip (word, value, &(l->localaddr));
     case CONTEXT_LNS:
         n = (struct lns *) item;
+        if (n->localrange) {
+            snprintf (filerr, sizeof (filerr),
+                    "'local ip range' and 'local ip' are mutually exclusive\n");
+            return -1;
+        }
         return set_ip (word, value, &(n->localaddr));
     default:
         snprintf (filerr, sizeof (filerr), "'%s' not valid in this context\n",
-- 
1.8.4.5

